Sprint 04

BezrukovAndrej opened the pull request:
Сделан Quiz согласно заданию спринта 4.

BezrukovAndrej committed:
Добавление согласно заданиям по спринту 4

BezrukovAndrej committed:
согласно заданию

BezrukovAndrej committed:
.

BezrukovAndrej committed:
.

BezrukovAndrej committed:
.

mmbabaev commented:
Также шрифты не меняются в приложении. В учебнике допущена ошибка, поэтому этот пункт необязателен. 
На скрине показал как нужно сделать: 
1) Шрифты добавлены в Group (серая папка)
2) Шрифтам проставлена галочка Target Membership  (справа)

<img width="1356" alt="Screen Shot 2022-11-17 at 20 15 01" src="https://user-images.githubusercontent.com/12600807/202513128-d74c5681-7a76-4778-ae5b-0fbcd44f0893.png">

Полезная статья, которая помогла мне разобраться в свое время: 
https://codewithchris.com/common-mistakes-with-adding-custom-fonts-to-your-ios-app/

mmbabaev commented:
Обязательно исправить: 
- рамка картинки не округленная
- рамка не пропадает после переключения вопроса
- цвет фона неправильный (нужен black из добавленных цветов)

mmbabaev commented:
Если быстро нажимать одну и ту же кнопку, то срабатывают лишние нажатия, пока мы ожидаем следующего вопроса. Чтобы этого не было, можно использовать свойство isEnabled. При начале ожидания "выключать" кнопку, после ожидания "включать" https://developer.apple.com/documentation/uikit/uicontrol/1618217-isenabled

mmbabaev commented:
обязательно: слово вопрос написано неверно

BezrukovAndrej committed:
Исправлено:
1) рамка картинки не округленная
2) рамка не пропадает после переключения вопроса
3) цвет фона неправильный (нужен black из добавленных цветов)
4)Если быстро нажимать одну и ту же кнопку, то срабатывают лишние нажатия, пока мы ожидаем следующего вопроса. Чтобы этого не было, можно использовать свойство isEnabled. При начале ожидания "выключать" кнопку, после ожидания "включать
5)обязательно: слово вопрос написано неверно

BezrukovAndrej committed:
Исправлено:
1) рамка картинки не округленная
2) рамка не пропадает после переключения вопроса
3) цвет фона неправильный (нужен black из добавленных цветов)
4) если быстро нажимать одну и ту же кнопку, то срабатывают лишние нажатия, пока мы ожидаем следующего вопроса. Чтобы этого не было, можно использовать свойство isEnabled. При начале ожидания "выключать" кнопку, после ожидания "включать" 5) обязательно: слово вопрос написано неверно

BezrukovAndrej commented:
1) рамка картинки не округленная
2) рамка не пропадает после переключения вопроса
3) цвет фона неправильный (нужен black из добавленных цветов)
4)Если быстро нажимать одну и ту же кнопку, то срабатывают лишние нажатия, пока мы ожидаем следующего вопроса. Чтобы этого не было, можно использовать свойство isEnabled. При начале ожидания "выключать" кнопку, после ожидания "включать
5)обязательно: слово вопрос написано неверно

mmbabaev commented:
<img width="638" alt="Screen Shot 2022-11-19 at 12 29 52" src="https://user-images.githubusercontent.com/12600807/202844487-589b7fd1-4d45-4993-b01c-b030a699d1d7.png">
сделайте пожалуйста здесь нижний отступ 0

BezrukovAndrej commented:
У моего приложение даже название такий констрейнов нету )

mmbabaev commented:
прошу прощения, перепутал скриншот. Вам нужно добавить нижний констрейнт к safe area c constant = 0 

Сейчас этого констрейнта нет и отступ разный на разных экранах

<img width="617" alt="Screen Shot 2022-11-19 at 13 39 39" src="https://user-images.githubusercontent.com/12600807/202846639-ad5deca6-3a03-468f-8ca3-2d2c79a87943.png">
